### DISCUSSION TITLE
Use Markdown link references

### DISCUSSION THREAD

**Discussion by arhohuttunen** [miniscruff/changie#602] [Ideas]
It would be nice to be able to use Markdown link references to keep the unrendered Markdown form of the changelog readable. This is recommended for example when following [Keep a Changelog](https://common-changelog.org/) or [Common Changelog](https://common-changelog.org/) formats.

For example:

```markdown
## [1.0.1] - 2019-08-24

### Fixed

- Prevent segmentation fault upon `close()`

## [1.0.0] - 2019-08-23

_Initial release._

[1.0.1]: https://github.com/owner/name/releases/tag/v1.0.1
[1.0.0]: https://github.com/owner/name/releases/tag/v1.0.0
```

**Comment by miniscruff** [comment]
Hmm, there is currently no method of adding a changelog footer like there is for headers. Nor would the header even include the data you would need to build this. It is an interesting idea and actually brings up some confusing parts of the config as I was looking around. The key of header is used in many places in similar but slightly different ways. I'm going to take a little time to see if I can fix the inconsistencies and include the option to support this feature.

Can't say when though, but I do like the idea.

**Reply by miniscruff** [reply]
Small update to this, after some thought I definitely want to overhaul, in a backwards compatible way, the different file and format strings in the configuration. I will make one or more tickets to handle this migration. Namely I think including changelog header/footer file and format configuration options that take in data for all releases. I believe that should be able to handle the markdown footer format structure via a range template.

**Reply by miniscruff** [reply]
Updating this one more time now that I have some stuff written here https://github.com/miniscruff/changie/discussions/610 which now includes a footer file and format for changelogs which will provide this functionality. No ETA yet but this may be the path forward.

**Comment by Blacksmoke16** [comment]
Could also make sense to allow them at the end of a specific version. E.g. 

```md
## [1.0.1] - 2019-08-24

### Added

- A shiny new feature ([#456])

### Fixed

- Prevent segmentation fault upon `close()` ([#123])

[#123]: https://github.com/owner/name/pulls/123
[#456]: https://github.com/owner/name/pulls/456

## [1.0.0] - 2019-08-23

_Initial release._

[1.0.1]: https://github.com/owner/name/releases/tag/v1.0.1
[1.0.0]: https://github.com/owner/name/releases/tag/v1.0.0
```

**Reply by miniscruff** [reply]
You can already build release note footers via https://changie.dev/config/#config-footerformat

Iirc dbt labs was doing some fancy stuff with this for authors and things.
